aws/iac/api_gateway.rego

```diff
@@ -0,0 +1,162 @@
+package rule
+
+# https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-restapi-endpointconfiguration.html
+
+#
+# PR-AWS-0202-CFR
+#
+
+default gateway_private = null
+
+aws_attribute_absence["gateway_private"] {
+    resource := input.Resources[i]
+    lower(resource.Type) == "aws::apigateway::restapi"
+    not resource.Properties.EndpointConfiguration.Types
+}
+
+aws_issue["gateway_private"] {
+    resource := input.Resources[i]
+    lower(resource.Type) == "aws::apigateway::restapi"
+    count(resource.Properties.EndpointConfiguration.Types) == 0
+}
+
+aws_issue["gateway_private"] {
+    resource := input.Resources[i]
+    lower(resource.Type) == "aws::apigateway::restapi"
+    type := resource.Properties.EndpointConfiguration.Types[_]
+    count([c | lower(type)== "private"; c:=1]) == 0
+}
+
+gateway_private {
+    lower(input.Resources[i].Type) == "aws::apigateway::restapi"
+    not aws_issue["gateway_private"]
+    not aws_attribute_absence["gateway_private"]
+}
+
+gateway_private = false {
+    aws_issue["gateway_private"]
+}
+
+gateway_private = false {
+    aws_attribute_absence["gateway_private"]
+}
+
+gateway_private_err = "AWS RestApi EndpointConfiguration types not contains private" {
+    aws_issue["gateway_private"]
+} else = "AWS RestApi EndpointConfiguration.Type is absent" {
+    aws_attribute_absence["gateway_private"]
+}
+
+gateway_private_metadata := {
+    "Policy Code": "PR-AWS-0202-CFR",
+    "Type": "IaC",
+    "Product": "AWS",
+    "Language": "AWS Cloud formation",
+    "Policy Title": "AWS RestApi EndpointConfiguration types not contains private",

```

Comment:
put a more obvious title for the policy.
for example: `API Gateway should have API Endpoint type as private and not exposed to internet`